### ISSUE TITLE
Utilize printing system in code generation

### ISSUE THREAD

**Issue by goerz** [mabuchilab/QNET#36]
Significant parts of the QSD code converter could be rewritten to make use of the printing system. Specifically, generating valid variable names from labels of "atomic" Expression, and generating infix expressions for operations can be handled through a `Printer`.

It might be useful to add a `_code_` method to `Expression` (akin to `_ascii_`, `_tex_`, etc.) that does some generic code generation (under the assumption that relevant languages such as C/C++, Fortran, Julia, are all pretty similar in terms of what acceptable variable names are and how expressions look.

This depends on #35

**Comment by goerz** [comment]
QSD code generation has been moved to a separate package, so this is no longer relevant to QNET
